apps/SparseMatrix/sparse_matrix_test.cu

```diff
@@ -166,6 +214,122 @@ TEST(Apps, SparseMatrix)
     spmat.free();
 }
 
+
+TEST(Apps, SparseMatrixQuery)
+{
+    using namespace rxmesh;
+
+    // Select device
+    cuda_query(0);
+
+    // generate rxmesh obj
+    std::string  obj_path = STRINGIFY(INPUT_DIR) "cube.obj";
+    RXMeshStatic rxmesh(obj_path);
+
+    uint32_t num_vertices = rxmesh.get_num_vertices();
+
+    const uint32_t threads = 256;
+    const uint32_t blocks  = DIVIDE_UP(num_vertices, threads);
+
+    SparseMatInfo<int> spmat(rxmesh);
+    spmat.set_ones();
+
+    LaunchBox<threads> launch_box;
+    rxmesh.prepare_launch_box(
+        {Op::VV}, launch_box, (void*)sparse_mat_query_test<threads>);
+
+    sparse_mat_query_test<threads>
+        <<<launch_box.blocks,
+           launch_box.num_threads,
+           launch_box.smem_bytes_dyn>>>(rxmesh.get_context(), spmat);
+
+    std::vector<uint32_t> h_result(spmat.m_nnz_entry_size);
+    CUDA_ERROR(cudaMemcpy(h_result.data(),
+                          spmat.m_d_val,
+                          spmat.m_nnz_entry_size * sizeof(int),
+                          cudaMemcpyDeviceToHost));
+
+    std::vector<uint32_t> h_ref(spmat.m_nnz_entry_size, 2);
+
+    for (uint32_t i = 0; i < spmat.m_nnz_entry_size; ++i) {
+        EXPECT_EQ(h_result[i], h_ref[i]);
+    }
+
+    spmat.free();
+}
+
+TEST(Apps, SparseMatrixEdgeLen)
+{
+    using namespace rxmesh;
+
+    // Select device
+    cuda_query(0);
+
+    // generate rxmesh obj
+    std::string  obj_path = STRINGIFY(INPUT_DIR) "dragon.obj";
+    RXMeshStatic rxmesh(obj_path);
+
+    uint32_t num_vertices = rxmesh.get_num_vertices();
+
+    const uint32_t threads = 256;
+    const uint32_t blocks  = DIVIDE_UP(num_vertices, threads);
+
+    std::vector<std::vector<float>> Verts;
+    auto coords = rxmesh.get_input_vertex_coordinates();
+
+    int* d_arr_ones;
+
+    std::vector<uint32_t> init_tmp_arr(num_vertices, 1);
+    CUDA_ERROR(cudaMalloc((void**)&d_arr_ones, (num_vertices) * sizeof(int)));
```

Comment:
This memory is being allocated but it is not freed

Reply:
Yeah. I just found a lot of things are not freed. I'll fix them as well.